openreview/tools.py

```diff
@@ -65,6 +65,32 @@ def create_profile(client, email, first, last, middle = None, allow_duplicates =
     else:
         return profile
 
+def get_preferred_name(client, group_id):
+    '''
+    Returns a string representing the user's preferred name, if available,
+    or the first listed name if not available.
+
+    Accepts emails or tilde ids.
+    '''
+
+    profile = client.get_profile(group_id)
+    names = profile.content['names']
+    preferred_names = [n for n in names if n.get('preferred', False)]
+    if preferred_names:
+        primary_preferred_name = preferred_names[0]
+    else:
+        primary_preferred_name = names[0]
+
+    name_parts = []
+    if primary_preferred_name.get('first'):
```

Comment:
you could do:

```
primary_preferred_name.get('first', '')
```

and no need to use an if.

Reply:
if you do this, and there's no middle name, the list that gets passed into the join() function will be ["First", "", "Last"], which will put two spaces between "First" and "Last". This was the best way I could come up with to get around that